scripts/task/KvToRdsMigration.php

```diff
@@ -0,0 +1,274 @@
+use oat\oatbox\extension\script\ScriptAction;
+use oat\taoDelivery\model\execution\ServiceProxy;
+use oat\taoOutcomeRds\model\RdsResultStorage;
+use oat\taoResultServer\models\classes\implementation\ResultServerService;
+
+class KvToRdsMigration extends ScriptAction
+{
+    use OntologyAwareTrait;
+
+    /** @var RdsResultStorage $kvStorageService */
+    protected $rdsStorage;
+
+    /** @var \taoResultServer_models_classes_ReadableResultStorage $kvStorageService */
+    protected $kvStorage;
+
+    /** @var array  */
+    protected $deliveryExecutions = [];
+
+    /** @var \common_report_Report */
+    protected $report;
+
+    /**
+     * Provide a "force" flag to apply the migration
+     *
+     * @return array
+     */
+    protected function provideOptions()
+    {
+        return [
+            'force' => [
+                'prefix' => 'f',
+                'longPrefix' => 'force',
+                'required' => false,
+                'default' => true,
+                'flag' => true,
+                'description' => 'Apply the migration to Result storage.',
+            ]
+        ];
+    }
+
+    /**
+     * Run the command.
+     *
+     * Get current result storage as key value storage
+     * Retrieve result $callIds and migrate deliveryExecution
+     * Migrate the result storage to rdsStorage
+     *
+     * @return \common_report_Report The report of the migration
+     * @throws \common_Exception
+     */
+    protected function run()
+    {
+        $this->kvStorage = $this->getCurrentKvResultStorage();
+
+        $this->rdsStorage = $this->getServiceLocator()->get(RdsResultStorage::SERVICE_ID);
+
+        $this->report = \common_report_Report::createInfo('Starting migration...');
+
+        foreach ($this->kvStorage->getAllCallIds() as $callId) {
+            foreach ($this->kvStorage->getVariables($callId) as $variables) {
+                $this->migrateDeliveryExecution($callId, $variables);
+            }
+        }
+
+        $this->setCurrentResultStorageToRdsStorage();
+
+        $this->report->add(\common_report_Report::createSuccess(count($this->deliveryExecutions) . ' delivery executions migrated.'));
+        if ($this->isDryrun()) {
+            $this->report->add(\common_report_Report::createFailure(
+                'The migration has not been applied because of dryrun mode. Use --force to really run the migration'
+            ));
+        }
+        return $this->report;
+    }
+
+    /**
+     * Migrate delivery execution variables
+     *
+     * Register the delivery execution if does not exist
+     * - If not then register related delivery & testtaker
+     * Migrate delivery execution variables
+     *
+     * @param $callId
+     * @param array $variables
+     * @throws \common_exception_Error
+     */
+    protected function migrateDeliveryExecution($callId, array $variables)
+    {
+        $deliveryExecutionIdentifier = $variables[0]->deliveryResultIdentifier;
```

Comment:
Is it safe to always call `$variables[0]` straight away, or would it be good to add an `isset($variables[0])` check for this?